content/academy/puppeteer_playwright/executing_scripts.md

```diff
@@ -0,0 +1,155 @@
+---
+title: III - Executing scripts
+description: Understand the two different contexts which your code can be run, and how to run custom scripts in the context of the browser.
+menuWeight: 7.3
+paths:
+    - puppeteer-playwright/executing-scripts
+---
+
+# [](#executing-scripts) Executing scripts
+
+An important concept to understand when dealing with headless browsers is the **context** in which your code is being run. For example, if you try to use the native `fs` Node.js module (used in the previous lesson) while running code in the context of the browser, errors will be thrown saying that it is undefined. Similarly, if you are trying to use `document.querySelector()` or other browser-specific functions in the server-side Node.js context, errors will also be thrown.
```

Comment:
This is super important and a source of a lot of confusion. Maybe we could reuse some of the content [from here](https://help.apify.com/en/articles/3195646-when-to-use-puppeteer-scraper) to explain it even more.

Reply:
I really like the diagram there, so I'll make a new one that fits our case

Reply:
Working on the diagram